homeassistant/components/stream/core.py

```diff
@@ -3,45 +3,54 @@
 
 import asyncio
 from collections import deque
-import io
-from typing import TYPE_CHECKING, Any, Callable
+import datetime
+from typing import Any, Callable
 
 from aiohttp import web
+import async_timeout
 import attr
 
 from homeassistant.components.http import HomeAssistantView
 from homeassistant.core import HomeAssistant, callback
 from homeassistant.helpers.event import async_call_later
 from homeassistant.util.decorator import Registry
 
-from .const import ATTR_STREAMS, DOMAIN
-
-if TYPE_CHECKING:
-    import av.container
-    import av.video
+from .const import ATTR_STREAMS, DOMAIN, MIN_SEGMENT_DURATION
 
 PROVIDERS = Registry()
 
 
-@attr.s
```

Comment:
The presentation of the refactoring here is actually a bit misleading. The previous StreamBuffer and the new Part are not really related.
Previously, StreamBuffer was defined in .core and only used internally in the SegmentBuffer class. I thought the class naming and hierarchy was a bit confusing (leading to things like self._segment.segment) and it was easier to remove the StreamBuffer class and place those attributes (segment, output, vstream, astream) directly into the SegmentBuffer class (renamed as SegmentBuffer._memory_file,SegmentBuffer._av_output,SegmentBuffer._output_video_stream,SegmentBuffer._output_audio_stream).